tests/ssg_test_suite/rule.py

```diff
@@ -399,30 +405,32 @@ def _get_rule_scenarios(self, rule):
                 templated_test_scenarios.keys())
         all_tests.update(templated_test_scenarios)
         all_tests.update(local_test_scenarios)
+        return all_tests
 
-        # Filter out everything except the shell test scenarios.
-        # Other files in rule directories are editor swap files
-        # or other content than a test case.
-        allowed_scripts = filter(lambda x: x.endswith(".sh"), all_tests)
-        content_mapping = {x: all_tests[x] for x in allowed_scripts}
-
+    def _get_rule_test_content(self, rule):
+        all_tests = self._load_all_tests(rule)
         scenarios = []
-        for script, script_contents in content_mapping.items():
-            scenario = Scenario(script, script_contents)
-            scenario.override_profile(self.scenarios_profile)
-            if (scenario.matches_regex(self.scenarios_regex) and
-                    scenario.matches_platform(self.benchmark_cpes)):
-                scenarios.append(scenario)
-        return scenarios
-
-    def _get_scenarios_by_rule_id(self, rules_to_test):
-        scenarios_by_rule_id = dict()
+        other_content = dict()
+        for file_name, file_content in all_tests.items():
+            # if the file name matches this regex, the file is a scenario
+            if re.search(r'.*\.[^.]*\.sh$', file_name):
+                scenario = Scenario(file_name, file_content)
+                scenario.override_profile(self.scenarios_profile)
+                if (scenario.matches_regex(self.scenarios_regex) and
+                        scenario.matches_platform(self.benchmark_cpes)):
```

Comment:
maybe extract that ugly condition, so one could read `if self._scenario_wanted_and_applicable(scenario):`